src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -112,10 +113,38 @@ ErrorIfRelationIsAKnownShard(Oid relationId)
 	}
 
 	const char *relationName = get_rel_name(relationId);
+
 	ereport(ERROR, (errmsg("relation \"%s\" is a shard relation ", relationName)));
 }
 
 
+/*
+ * ErrorIfChangingKnownShard errors out if the relation with relationId is
+ * a known shard and manual changes on known shards are disabled. This is
+ * valid for only non-citus (external) connections.
+ */
+void
+ErrorIfChangingKnownShard(Oid relationId)
+{
+	if (IsCitusInitiatedRemoteBackend() || EnableManualChangesToShards)
+	{
+		return;
+	}
+
+	/* search the relation in all schemas */
+	bool onlySearchPath = false;
+	if (RelationIsAKnownShard(relationId, onlySearchPath))
+	{
+		const char *relationName = get_rel_name(relationId);
+		ereport(ERROR, (errmsg("relation \"%s\" is a shard relation ", relationName),
```

Comment:
Minor: the wording of the errmsg feels slightly strange. Maybe "a shard relation" -> "a shard of a distributed table"

It could also be nice to say what went wrong:

`cannot modify \"%s\" because it is a shard of a distributed table`